### ISSUE TITLE
Holding Strong Reference to Client

### ISSUE THREAD

**Issue by calebkleveter** [vapor-community/Imperial#26]
Somewhere we appear to be holding onto `Client` when the `container` is released, cause us to hit a fatalError in Vapor:

```
Fatal error: If you encounter this error, you are holding on to a client after de-initializing your Application. This is usually a bad idea.: file /tmp/build_769e8eccc58c3bc6a5daf7c3903d82c0/.build/checkouts/vapor.git-4085136337785951737/Sources/Vapor/Client/FoundationClient.swift, line 16
```

This is a design issue on our side.

**Comment by 0xTim** [comment]
This should be fixed by vapor/vapor#1896

**Comment by 0xTim** [comment]
This should be fixed in Vapor 4
